main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -222,12 +228,19 @@ object Weeder2 {
 
     private def visitModuleDecl(tree: Tree)(implicit sctx: SharedContext, flix: Flix): Validation[Declaration.Namespace, CompilationMessage] = {
       expect(tree, TreeKind.Decl.Module)
+      val qname = pickQName(tree) match {
+        case Some(name) => name
+        case None =>
+          // Parser has reported an error so return some possibly invalid name
+          val name = text(tree)
```

Comment:
everywhere we use `text` for error handling we could consider using `flix` or `Symbol` to generate a unique name